### PR TITLE
handle detached array in XRView

### LINKED ISSUE #26153: Handle detached arrays in XRView
There are a bunch of places in the WebXR spec where arrays are returned. If you call `.transfer()` on these, their buffers are "detached", and need to be recomputed. You can check if a buffer is detached by calling `is_null()` on it.

We handle most of these cases, except for [`XRView.projectionMatrix`](https://immersive-web.github.io/webxr/#xrview-obtain-the-projection-matrix)

We need to store the projection matrix on the XRView and use a null check in `ProjectionMatrix` to recompute it if necessary:

https://github.com/servo/servo/blob/5e91e48293406cebf6fef4ed03bcae568021c348/components/script/dom/xrview.rs#L83

### PR DESCRIPTION
**Pull request by daemon1024**
Recompute XRView.projectionmatrix when the buffer is detached
---

- [x] `./mach build -d` does not report any errors
- [x] `./mach test-tidy` does not report any errors
- [x] These changes fix #26153 (GitHub issue number if applicable)

- [ ] There are tests for these changes OR
- [ ] These changes do not require tests because ___

### CODE REVIEW

File: components/script/dom/xrview.rs [hunk 1]
```diff
@@ -24,17 +25,24 @@ pub struct XRView {
     proj: Heap<*mut JSObject>,
     #[ignore_malloc_size_of = "mozjs"]
     view: Heap<*mut JSObject>,
+    viewsrc: Vec<f32>,
```
Comment (Manishearth) [inline components/script/dom/xrview.rs:28]: let's call this `proj_array`, here and elsewhere in the code

```suggestion
    proj_array: Vec<f32>,
```

File: components/script/dom/xrview.rs [hunk 2]
```diff
@@ -55,10 +63,14 @@ impl XRView {
         let transform = pose.pre_transform(&offset);
         let transform = XRRigidTransform::new(global, cast_transform(transform));
 
-        let ret = reflect_dom_object(
-            Box::new(XRView::new_inherited(session, &transform, eye)),
-            global,
-        );
+        let projm = view.projection.to_row_major_array();
+        let xrv = Box::new(XRView::new_inherited(
```
Comment (Manishearth) [inline components/script/dom/xrview.rs:67]: We typically keep the reflect_dom_object in one statement

File: components/script/dom/xrview.rs [hunk 3]
```diff
@@ -55,10 +63,14 @@ impl XRView {
         let transform = pose.pre_transform(&offset);
         let transform = XRRigidTransform::new(global, cast_transform(transform));
 
-        let ret = reflect_dom_object(
-            Box::new(XRView::new_inherited(session, &transform, eye)),
-            global,
-        );
+        let projm = view.projection.to_row_major_array();
```
Comment (Manishearth) [inline components/script/dom/xrview.rs:66]: this is already done below, you should just shift this line down

**Review by Manishearth — APPROVED** [review]
Please squash your commits and we can land this!

**Comment by Manishearth** [comment]
@bors-servo r+

thanks!

**Comment by jdm** [comment]
@bors-servo retry
* #26192 #25513

**Comment by jdm** [comment]
@bors-servo retry
* #26192

**Comment by jdm** [comment]
@bors-servo retry
* #21067

**Comment by jdm** [comment]
@bors-servo retry
* #25513

**Pull request merged**